### DISCUSSION TITLE
TC_STAT CTC output does not contain storm identifier

### DISCUSSION THREAD

**Discussion by mollybsmith-noaa** [dtcenter/METplus#2608] [Configuration] [answered]
Hi,

I've been checking our TC verification workflow prior to this year's HFIP, and I noticed that when you run TC_STAT with `-by CYCLONE -out_line_type CTC`, there is indeed a line in the resulting stat file for each cyclone, but there is no indication which line belongs to which cyclone.  Would it be possible to include the cyclone number somewhere in the CTC output? Otherwise the stratified stats are largely useless.

Also, when you try to use `-by lead`, TC_STAT seems to ignore whatever type of output you want and just gives you CTS. I think something may be hardcoded in a loop there.

Molly

**Comment by JohnHalleyGotway** [comment, marked as answer]
@mollybsmith-noaa, I ran the following job on the output from the MET unit tests to try this out:
```
./tc_stat -lookin test_output/tc_pairs -job rirw -line_type TCMPR -by CYCLONE -out_line_type CTC
```
By default, output is written to the screen and that output DOES include the `CYCLONE` value, as well as any other `-by` columns you may add:
```
DEBUG 2: Processing Job 1: -job rirw -line_type TCMPR -rirw_time 240000 -rirw_exact true -rirw_thresh >=30.0 -rirw_window 000000 000000 -by CYCLONE -out_line_type CTC -out_alpha 0.05000 
JOB_LIST: -job rirw -line_type TCMPR -rirw_time 240000 -rirw_exact true -rirw_thresh >=30.0 -rirw_window 000000 000000 -by CYCLONE -out_line_type CTC -out_alpha 0.05000 
COL_NAME: ATIME  BTIME  AEXACT BEXACT ATHRESH BTHRESH WINDOW_BEG WINDOW_END CYCLONE TOTAL FY_OY FY_ON FN_OY FN_ON EC_VALUE
RIRW_CTC: 240000 240000 true   true   >=30.0  >=30.0  000000     000000     01          8     0     0     0     8      0.5
RIRW_CTC: 240000 240000 true   true   >=30.0  >=30.0  000000     000000     02        148     0     0     0   148      0.5
RIRW_CTC: 240000 240000 true   true   >=30.0  >=30.0  000000     000000     03         12     0     0     0    12      0.5
...
```
The issue arises when you use the `-out_stat ctc.stat` job command option to write output to a `.stat` file rather than the screen. The case-specific information (i.e. the `CYCLONE` number) does not appear in that output file by default.

The Stat-Analysis tool supports a job command option named `-set_hdr` that solves this type of problem. You can manually override header column values is case-specific values. Listed below is some documentation from [this section](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#jobs) of the MET User's Guide:

> When using the “-by” job command option, you may reference those columns in the “-set_hdr” job command options. For example, when computing statistics separately for each station, write the station ID string to the VX_MASK column of the output .stat output file:
> 
> -job aggregate_stat -line_type MPR -out_line_type CNT \
> -by OBS_SID -set_hdr VX_MASK OBS_SID -stat_out out.stat
> When using mulitple "-by" options, use "CASE" to reference the full string:
> -by FCST_VAR,OBS_SID -set_hdr DESC CASE -stat_out out.stat

However, when I tried adding `-set_hdr DESC CYCLONE` to the TC-Stat job, I got the following error message:
```
ERROR  : TCStatJob *TCStatJobFactory::new_tc_stat_job(const char *jobstring) -> unsupported job command options "-set_hdr DESC CYCLONE".
```

So what's really needed here is to enhance TC-Stat to port over support for the `-set_hdr` job command option from Stat-Analysis. Does that make sense? Do you agree that functionality would solve this issue? If so, I can write up a MET issue to document this change.

**Comment by mollybsmith-noaa** [comment]
Thanks, John. I do think that making “-set_hdr” work with TC Stat would fix
that issue!

Were you also able to stratify by CYCLONE,LEAD? When I tried that, the
-out_stat command didn’t properly produce a CTC stat file.

Thanks again for looking into this!
Molly

**Reply by JohnHalleyGotway** [reply]
Molly, OK, I'll write up the issue for this change. Yes, I did just test using `-by CYCLONE,LEAD -out_line_type CTC -out_stat ctc.stat` and the output file really does just contain a bunch of CTC lines. I don't see any spurious CTS lines showing up in the output. By default, both CTC and CTS lines are written to the output... so if do not set `-out_line_type CTC` to override that default, then you really will see CTS output lines.

**Reply by JohnHalleyGotway** [reply]
OK, please see dtcenter/MET#2911 which is an issue that describes what's needed. I'd recommend that we mark this discussion as being answered and move future discussion to comments on that issue.

**Comment by mollybsmith-noaa** [comment]
Will do. I’ll also check again with the lead time one to see what I did
wrong there. If you got the expected output then I probably had a typo or
something.

Thanks!
Molly
